presentation/src/main/java/com/knocklock/presentation/home/HomeViewModel.kt

```diff
@@ -36,29 +39,50 @@ class HomeViewModel @Inject constructor(
         .stateIn(
             scope = viewModelScope,
             started = SharingStarted.WhileSubscribed(5_000),
-            initialValue = LockScreen()
+            initialValue = LockScreen(),
         )
 
     val homeScreenUiState = combine(
         homeMenuList,
         tmpHomeScreen,
-        savedHomeScreenState
+        savedHomeScreenState,
     ) { menuList, tmpScreen, savedScreen ->
         HomeScreenUiState.Success(
             menuList = menuList.toImmutableList(),
-            lockScreen = if (tmpScreen is TmpScreenState.Custom) tmpScreen.screen else savedScreen
+            lockScreen = if (tmpScreen is TmpScreenState.Custom) tmpScreen.screen else savedScreen,
         )
     }.stateIn(
         scope = viewModelScope,
         started = SharingStarted.WhileSubscribed(5_000),
-        initialValue = HomeScreenUiState.Loading
+        initialValue = HomeScreenUiState.Loading,
     )
 
-    fun saveLockScreen() {
+    fun saveLockScreen(context: Context) {
         viewModelScope.launch {
             val value = tmpHomeScreen.value
             if (value is TmpScreenState.Custom) {
-                saveLockScreenCase(value.screen)
+                val uri = Uri.parse((value.screen.background as LockScreenBackground.LocalImage).imageUri)
+                if (uri.path == null) {
+                    return@launch
+                } else {
+                    val specificFile = File(uri.path).name
+                    val fileNames = context.filesDir.list()
+                    fileNames?.let { _ ->
+                        var index = 0
+                        while (index != fileNames.size) {
+                            if (fileNames[index] != specificFile) {
+                                val deleted = File(context.filesDir, fileNames[index]).delete()
+                                if (deleted) {
+                                    println("로그: File '${fileNames[index]}' has been removed.")
+                                } else {
+                                    println("로그: Failed to remove file '${fileNames[index]}'.")
+                                }
+                            }
+                            index++
+                        }
+                    }
```

Comment:
```suggestion
                    val specificFile = File(uri.path ?: return@launch)
                    context.filesDir.listFiles()?.forEach { file ->
                        if (file.name != specificFile.name) {
                            if (file.delete()) {
                                Log.d("HomeViewModel-File", "제거 성공")
                            } else {
                                Log.d("HomeViewModel-File", "$file 제거 실패")
                            }
                        }
                    }
```

위처럼 변경하면 어때요??

1) index 관리 필요 x
2) println은 릴리즈모드에서도 그래로 노출됨

Reply:
@jaeryo2357 야무지게 간결해졌당 ㅎ

Reply:
이거로 반영할꼐영

Reply:
@jaeryo2357 [c977b31](https://github.com/Knock-Lock/Knock-Lock/pull/177/commits/c977b313275e9f46ec65a27ef324e2ccb41d9a33) 반영했어용